OpenRA.Game/Graphics/PlayerColorRemap.cs

```diff
@@ -27,28 +27,33 @@ public static int GetRemapIndex(int[] ramp, int i)
 
 		public PlayerColorRemap(int[] ramp, Color c, float rampFraction)
 		{
-			var h = c.GetHue() / 360.0f;
-			var s = c.GetSaturation();
-			var l = c.GetBrightness();
-
-			// Increase luminosity if required to represent the full ramp
-			var rampRange = (byte)((1 - rampFraction) * l);
-			var c1 = Color.FromAhsl(h, s, Math.Max(rampRange, l));
-			var c2 = Color.FromAhsl(h, s, (byte)Math.Max(0, l - rampRange));
-			var baseIndex = ramp[0];
-			var remapRamp = ramp.Select(r => r - ramp[0]);
-			var rampMaxIndex = ramp.Length - 1;
-
-			// reversed remapping
-			if (ramp[0] > ramp[rampMaxIndex])
+			if (ramp.Any())
```

Comment:
Personal preference:
 - No LINQ
 - Invert statement to reduce nesting
 ```cs
if (ramp.Length == 0)
{
    remapColors = new Dictionary<int, Color>();
    return;
}
```
As a side bonus that minimizes the detected changes in this file.
(hardly relevant since this all goes away in a follow-up commit, but if you want commits to be usable individually...)